mlflow/sklearn/__init__.py

```diff
@@ -455,6 +460,323 @@ def load_model(model_uri):
+        for arg in call_arg_list:
+            if arg is not None and not np.isscalar(arg) and id(arg) in dataset_id_list:
+                dataset_name, run_id = self._pred_result_id_to_dataset_name_and_run_id[id(arg)]
+                break
+        else:
+            return False, None, None
+
+        metric_key = self.register_metric_info(
+            run_id, metric_name, dataset_name, call_command
+        )
+
+        return True, run_id, metric_key
+
+    def log_eval_metric(self, run_id, key, value):
+        # Note: if the case log the same metric key multiple times,
+        #  newer value will overwrite old value
+        client = mlflow.tracking.MlflowClient()
+        client.log_metric(
+            run_id=run_id,
+            key=key,
+            value=value
+        )
+        if self._metric_info_artifact_need_update[run_id]:
+            call_commands_list = []
+            for v in self._metric_api_call_info[run_id].values():
+                call_commands_list.extend(v)
+
+            call_commands_list.sort(key=lambda x: x[0])
+            dict_to_log = OrderedDict(call_commands_list)
+            client.log_dict(
+                run_id=run_id, dictionary=dict_to_log, artifact_file='metric_info.json'
+            )
+            self._metric_info_artifact_need_update[run_id] = False
+
+    def get_method_first_arg_value(self, method, call_pos_args, call_kwargs):
+        """
+        Get instance method first argument value (exclude the `self` argument).
+        """
+        if len(call_pos_args) >= 1:
+            return call_pos_args[0]
+        else:
+            first_arg_name = list(inspect.signature(method).parameters.keys())[1]
+            return call_kwargs.get(first_arg_name)
+
+
+_autolog_training_status = _AutologTrainingStatus()
+
+
+def _get_autolog_training_status():
+    """
+    Get the global `_AutologTrainingStatus` instance which holds information used in post-training
+    metric autologging. See doc of class `_AutologTrainingStatus` for details.
+    """
+    return _autolog_training_status
+
+
+_metric_api_excluding_list = [
+    'check_scoring', 'get_scorer', 'make_scorer'
+]
+
+
+def _get_metric_name_list():
+    """
+    Return metric function name list in `sklearn.metrics` module
+    """
+    from sklearn import metrics
+    metric_list = []
+    for metric_method_name in metrics.__all__:
+        # excludes plot_* methods
+        # exclude class (e.g. metrics.ConfusionMatrixDisplay)
+        metric_method = getattr(metrics, metric_method_name)
+        if metric_method_name not in _metric_api_excluding_list \
+                and not inspect.isclass(metric_method) \
+                and callable(metric_method) \
+                and not metric_method_name.startswith('plot_'):
+            metric_list.append(metric_method_name)
+    return metric_list
+
+
 @experimental
 @autologging_integration(FLAVOR_NAME)
 def autolog(
```

Comment:
As a follow-up PR, we should introduce a flag to control whether or not post-training metrics should be logged. This is particularly important for backwards compatibility with existing tools that add metrics to autologging runs (e.g. AutoML tools).

Reply:
Create followup ticket https://databricks.atlassian.net/browse/ML-16605

Reply:
Thanks!